frontend/pages/reviewRoles/view.tsx

```diff
@@ -64,53 +103,92 @@ export default function ReviewRoles() {
+          </Stack>
+        ))
+      : 'No entities assigned'
+  }, [formData])
+
   const listRoleDescriptions = useMemo(
     () =>
       reviewRoles.map((reviewRole, index) => (
         <Fragment key={reviewRole._id}>
           {selectedRole === index && (
             <>
-              <Stack spacing={2}>
-                <Box>
-                  <Typography color='primary' fontWeight='bold'>
-                    Description
-                  </Typography>
-                  <Typography>{reviewRole.description}</Typography>
-                </Box>
-                <Box>
-                  <Typography color='primary' fontWeight='bold'>
-                    System Role
-                  </Typography>
-                  {reviewRole.systemRole ? (
-                    <Typography>{toTitleCase(reviewRole.systemRole)}</Typography>
-                  ) : (
-                    <em>None</em>
-                  )}
-                </Box>
-                <Box>
-                  <Typography color='primary' fontWeight='bold'>
-                    Default entities
-                  </Typography>
-                  <List>
-                    {!reviewRole.defaultEntities ||
-                      (reviewRole.defaultEntities.length === 0 && <em>No entities assigned</em>)}
-                    {reviewRole.defaultEntities &&
-                      reviewRole.defaultEntities.map((entity) => (
-                        <ListItem key={entity}>
-                          <UserDisplay dn={entity} />
-                        </ListItem>
-                      ))}
-                  </List>
-                </Box>
-                <Button
-                  color='error'
-                  sx={{ width: 'max-content' }}
-                  variant='contained'
-                  onClick={handleOpenDeleteConfirmation}
-                >
-                  Delete role
-                </Button>
-              </Stack>
+              {!isEdit ? (
+                <Stack spacing={2}>
+                  <Box>
+                    <Typography color='primary' fontWeight='bold'>
+                      Description
+                    </Typography>
+                    <Typography>{formData?.description || 'No description'}</Typography>
+                  </Box>
+                  <Box>
+                    <Typography color='primary' fontWeight='bold'>
+                      System Role
+                    </Typography>
+                    <Typography>
+                      {formData?.systemRole ? getRoleDisplayName(formData?.systemRole, modelRoles) : 'No system role'}
+                    </Typography>
+                  </Box>
+                  <Box>
+                    <Typography color='primary' fontWeight='bold'>
+                      Default entities
+                    </Typography>
+                    <List>{displayReviewRoleDefaultEntities}</List>
+                  </Box>
+                  <Box display='flex'>
+                    <Box ml='auto'>
```

Comment:
I don't think we want this right-aligned at the bottom, I think we either leave this left-aligned as it was before, or we move it to the top of the form. 

We also only need one `<Box />` component